core/src/main/java/io/kestra/core/contexts/KestraContext.java

```diff
@@ -0,0 +1,101 @@
+package io.kestra.core.contexts;
+
+import io.kestra.core.models.ServerType;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.annotation.Context;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.env.Environment;
+import jakarta.annotation.PreDestroy;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Utility class for retrieving common information about a Kestra Server at runtime.
+ */
+public abstract class KestraContext {
```

Comment:
If everithing is implemented by the Initializer, this may be an interface.

Reply:
Well, ideally, I would also prefer to have an interface, but I've used an abstract class to provide encapsulation for the SINGLETON.